### DISCUSSION TITLE
Custom MFXListCell requires Node in Node

### DISCUSSION THREAD

**Discussion by ponchoboob** [palexdev/MaterialFX#374] [Q&A]
First of all, great library! big fan

I recently explored customizing the content of a MFXListView's ListCell. It's somewhat different from the typical approach used with JavaFX ListCell. My inquiry revolves around the use of StringConverter and the default Label. In MFXListCell, there's a distinction based on whether the model data is of type Node. For customizing my ListCell, it seems necessary to encapsulate the model data within another model class that extends from HBox or any other preferred control. Given that MFXListCell inherently functions as an HBox, I'm curious about the rationale behind the additional Label that is automatically included as a child within the MFXListCell (which is essentially an HBox). Of course I get provide a StringConverter that returns an emptry string. But that label is still there as a child of the MFXListCell's HBox.

kind regards.

**Comment by palexdev** [comment]
I did it this way for two main reasons:
1) To hide the methods and properties of the `Label`. Since the cells' content depend on the item and index, users should not be allowed to do something like this `cell.setText(...)`. HBox because the layout is automatic and in most cases in UI, cells' contents are arranged horizontally
2) Having a wrapper container also allows to have as many nodes as desired inside the cell. If you are building a complex cell with multiple nodes it's as simple as creating the node, maybe store its insurance if needed, and then just do `getChildren().add(node)`

For your specific use case, if the model data type are nodes, then you just need a custom cell implementation.
If you don't need the label it's better to create a new class that `implements` `Cell`.
If you do need the label then you can simply extend the current implementation and add the node to the HBox every time the `itemProperty` changes

**Reply by ponchoboob** [reply]
Hi, thank you for your fast reply and explanation.
kind regards.

**Reply by ponchoboob** [reply]
Actually my model data are not Nodes. I know that the basic layout of MFXListCell is a HBox. But I was curious about the extra label. When I first wrote a MFXListCell factory with an ImageView I always had the class name of the model class as the label. That was confusing me because my actual cell layout was just an ImageView.  So when the model class is just a POJO then the label of MFXListCell is set automatically. When writing the MFXListCell factory the label actually is not needed because you do the layout yourselfe and put it into the hbox. So you either provide a StringConverter that returns an empty string or you provide a model class that inherits a javafx control.

kind regards
